typings/built-in-glamorous-components.d.ts

```diff
@@ -15,426 +16,670 @@ import {CSSProperties} from './css-properties'
   >
   Tbody: React.StatelessComponent<
-    CSSProperties &
+    CSSPropertiesCompleteSingle &
       ExtraGlamorousProps &
       React.HTMLProps<HTMLTableSectionElement>
   >
   Td: React.StatelessComponent<
-    CSSProperties &
+    CSSPropertiesCompleteSingle &
       ExtraGlamorousProps &
       React.HTMLProps<HTMLTableDataCellElement>
   >
   Textarea: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLTextAreaElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLTextAreaElement>
   >
   Tfoot: React.StatelessComponent<
-    CSSProperties &
+    CSSPropertiesCompleteSingle &
       ExtraGlamorousProps &
       React.HTMLProps<HTMLTableSectionElement>
   >
   Th: React.StatelessComponent<
-    CSSProperties &
+    CSSPropertiesCompleteSingle &
       ExtraGlamorousProps &
       React.HTMLProps<HTMLTableHeaderCellElement>
   >
   Thead: React.StatelessComponent<
-    CSSProperties &
+    CSSPropertiesCompleteSingle &
       ExtraGlamorousProps &
       React.HTMLProps<HTMLTableSectionElement>
   >
   Time: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLTimeElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLTimeElement>
   >
   Title: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLTitleElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLTitleElement>
   >
   Tr: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLTableRowElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLTableRowElement>
   >
   Track: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLTrackElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLTrackElement>
   >
   U: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLElement>
   >
   Ul: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLUListElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLUListElement>
   >
   Var: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLElement>
   >
   Video: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLVideoElement>
+    CSSPropertiesCompleteSingle &
+      ExtraGlamorousProps &
+      React.HTMLProps<HTMLVideoElement>
   >
   Wbr: React.StatelessComponent<
-    CSSProperties & ExtraGlamorousProps & React.HTMLProps<HTMLElement>
+    CSSPropertiesCompleteSingle &
```

Comment:
So being unable to use `':active'` or other pseudo properties works as intended?
I mean,

```ts
const props = {
  ':active': {
    color: 'purple'
  }
};

<glamorous.A {...props} />
```

Is invalid?

Reply:
hmm, I hadn't really though about spreading the css props on.

🤔 That is valid, but as I understood it the purpose of that api was to make it possible to simple add css properties like follows;

```ts
<glamorous.A color="purple" />
```

And for the pattern you're describing do something like follows
```
const cssProps = {
  ':active': {
    color: 'purple'
  }
};
<glamorous.A css={props} />
```

I'm leaning towards making the types simpler here to reflect the intended use. As otherwise it makes the types less useful elsewhere.

Really appreciate your feedback and taking the time to go over these prs and even skimming the code 💖.

Reply:
We could add `CSSPropertiesPseudo` to the Intersection which would allow the case you've raised to work. But it would still fall down on things like

const props = {
  '> span': {
    color: 'purple'
  }
};

<glamorous.A {...props}>hello <span>world</span></glamorous.A>

Reply:
Thank yo for your kind answer!
I think I fully get the point what you want to say, and I agree with it to some distance. I will approve PR.

Reply:
I've added test cases for the pseudo props as part of this pr as well, cheers.

https://github.com/luke-john/glamorous/blob/a821469f2cad3c8511bbdfc12d2c807d6cc4d08f/test/glamorous.test.tsx#L611-L621